proto/distributed_cache.proto

```diff
@@ -1,14 +1,11 @@
 syntax = "proto3";
 
+import "proto/resource.proto";
+
 package distributed_cache;
 
 message Isolation {
-  enum CacheType {
-    UNKNOWN_TYPE = 0;
-    ACTION_CACHE = 1;
-    CAS_CACHE = 2;
-  }
-  CacheType cache_type = 1;
+  resource.CacheType cache_type = 1;
```

Comment:
I think a simpler fix is to change the enum values in resource.proto to match those in distributed_cache.proto since I don't believe the values from resource.proto are persisted anywhere (yet).

Reply:
It looks like `GetCacheMetadata` API is already using the current enum values in resource.proto, but it's fine if this API is temporarily broken during the release. I don't think anybody is going to notice.

Reply:
the ScoreCard persists cache_type to disk, so if we change the scorecard to use resource.proto then we'd probably need to have backwards-compatibility logic in place for old cache scorecards. If the distributed cache protos aren't persisted (but just used for communication) then migrating the distributed cache might avoid a bit of cruft. But if it would be too complicated to do that migration then having the backwards compatibility for the scorecard instead instead seems OK

Reply:
Oh lovely, didn't realize there was a third cache_type enum... 

There are customers with on-prem deployments of the distributed cache so I'd rather not force a migration for them.

I think it's better if we put in some temporary cruft for the scorecard and then after a while (a month?) remove support for the old field since I think it's unlikely people would be looking up the scorecard for older invocations.

Reply:
+1 to what @vadimberezniker said if we can confirm those are not written yet

Reply:
Okay - agreed that the easiest option forward is to change the ordering of the enum values in resource.proto. I will make the change to do that, and then will break up this PR into two parts - 1. Delete the two enums that match the new ordering in resource.proto. Will be backwards compatible 2. Delete the enums that don't match the ordering in resource.proto. Will need a temporary field for the scorecards to be backwards compatible